rpc/jsonrpc/server/http_server.go

```diff
@@ -53,10 +53,11 @@ func DefaultConfig() *Config {
 func Serve(listener net.Listener, handler http.Handler, logger log.Logger, config *Config) error {
 	logger.Info("serve", "msg", log.NewLazySprintf("Starting RPC HTTP server on %s", listener.Addr()))
 	s := &http.Server{
-		Handler:        RecoverAndLogHandler(maxBytesHandler{h: handler, n: config.MaxBodyBytes}, logger),
-		ReadTimeout:    config.ReadTimeout,
-		WriteTimeout:   config.WriteTimeout,
-		MaxHeaderBytes: config.MaxHeaderBytes,
+		Handler:           RecoverAndLogHandler(maxBytesHandler{h: handler, n: config.MaxBodyBytes}, logger),
+		ReadTimeout:       config.ReadTimeout,
+		ReadHeaderTimeout: config.ReadTimeout,
```

Comment:
How is this related? I think we should keep changes scoped just to the ABCI++ work for now.

Reply:
It also looks like, from the Go docs, this is not necessary, since ReadHeaderTimeout takes the value of ReadTimeout if not set. 

https://pkg.go.dev/net/http#Server

Reply:
Yeah it was picked up by the l'Inter I believe

Reply:
Hmm, mind pointing me toward the lint error? Not sure why we have a linter recommending we set this.